File: LeetCode/01two_sum/Solution.java
/**
*Input: numbers={0,4,3,0}, target=0
*Output: index1=1, index2=2
*/
import java.util.*;

public class Solution {
	//numbers= 5,75,25  target=100
    public int[] twoSum(int[] numbers, int target) {
    	Map<Integer, Integer> map=new HashMap<>();
        int length=numbers.length; //length=3
        for(int i=0;i<length;i++){
        	int x=numbers[i]; //95
        	//search for index2
        	if(map.containsKey(target-x)){
        		return new int[] {map.get(target-x)+1,i+1 };
        	}
        	map.put(x,i);
    	}   
    	throw new IllegalArgumentException("No Solution");
	}
}

File: LeetCode/01two_sum/Solution.cpp
#include <unordered_map>
#include <vector>

using std::vector;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
	vector<int> solution(2,0);
	for (int i=0; i<nums.size(); i++) {
	    for (int j=i; j<nums.size(); j++) {
		if (target - nums[i] == nums[j]) {
		    solution={i, j};
		}
	    }
	}
	return solution;
    }
};

#include <iostream> 

int main() {
    Solution* s = new Solution();
    vector<int> nums = {2, 7, 11, 15};
    auto rst = s->twoSum(nums, 9);
    for (auto i:rst) {
	std::cout << i << "";
    }
    std::cout << "\n";
}

